### PR TITLE
ENH: Fix Python 3.13 test failures & enable CI

### PR DESCRIPTION
**Pull request by lysnikolaou**
x-ref #58734

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -4948,7 +4948,12 @@ cpdef to_offset(freq, bint is_period=False):
     if result is None:
         raise ValueError(INVALID_FREQ_ERR_MSG.format(freq))
 
-    if is_period and not hasattr(result, "_period_dtype_code"):
+    try:
+        has_period_dtype_code = hasattr(result, "_period_dtype_code")
+    except ValueError:
```
[the remark this answers is not in the document]
Comment (lysnikolaou) [reply pandas/_libs/tslibs/offsets.pyx:4953]: The error is that the `_period_dtype_code` property raises `ValueError` in `CustomBusinessDay`. In 3.12 and older, Cython was silently ignoring exception and `hasattr` was returning `False`. In 3.13 the exception is not ignored anymore. More info on https://github.com/cython/cython/issues/6251#issuecomment-2182781972.

**Comment by lysnikolaou** [comment]
I'm marking as ready for review, since I don't think there's any more outstanding issues for 3.13.

**Review by lithomas1 — APPROVED** [review]
This LGTM pending green.

Thanks for picking this up!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lysnikolaou and @lithomas1

**Comment by lithomas1** [comment]
@meeseeksdev backport 2.2.x

**Pull request merged**